getting_started/kubernetes-overview.adoc

```diff
@@ -0,0 +1,105 @@
+
+|`container-runtime`
+|`container-runtime` runs containers on a host operating system. You must install container-runtime on each node so that pods can run on the node.
+
+|Persistent storage
+|Stores the data even after the device is shut  off. Kubernetes uses persistent volumes to store the application data.
+
+|`container-registry`
+|Stores and accesses the container images.
+
+|Pod
+|The pod is the smallest logical unit in Kubernetes. A pod contains one or more containers to run in a worker node.
+|===
+
+A cluster is a single computational unit consisting of multiple nodes in a cloud environment. A Kubernetes cluster includes a control plane and worker nodes. You can run Kubernetes containers across various machines and environments. The control plane node controls and maintains the state of a cluster. You can run the Kubernetes application by using worker nodes. You can use the Kubernetes namespace to differentiate cluster resources in a cluster. Namespace scoping is applicable for resource objects, such as deployment, service, and pods. You can not use namespace for cluster-wide resource objects such as `storageclass`, nodes, and persistent volumes.
+
+.Kubernetes cluster overview
+image::247_OpenShift_Kubernetes_Overview-1.png[]
+
+A custom resource is an extension of the Kubernetes API. You can customize Kubernetes clusters by using custom resources. Operators are software extensions which manage applications and their components with the help of custom resources. Kubernetes uses a declarative model when you want a fixed desired result while dealing with cluster resources. By using Operators, Kubernetes defines its states in a declarative way. You can modify the Kubernetes cluster resources by using imperative commands.
+An Operator acts as a control loop which continuously compares the desired state of resources with the actual state of resources and puts actions in place to bring reality in line with the desired state.
+
+.Kubernetes Resources
+[cols="1,2",options="header"]
+|===
+|Resource |Purpose
+
+|Service
+|Kubernetes uses services to expose a running application on a set of pods.
+
+|ReplicaSets
+|Kubernetes uses ReplicaSets to maintain the constant pod number.
+
+|Deployment
+|A resource object that maintains the life cycle of an application.
+|===
+
+Kubernetes is a core component of an {product-title}. You can use {product-title} for developing and running containerized applications. With its foundation in Kubernetes, the {product-title} incorporates the same technology that serves as the engine for massive telecommunications, streaming video, gaming, banking, and other applications. You can extend your containerized applications beyond a single cloud to on-premise and multi-cloud environments by using the {product-title}.
+
+.Architecture of Kubernetes
+image::247_OpenShift_Kubernetes_Overview-2.png[]
+
```

Comment:
This "before getting started chunk of content" could be its own conceptual module called Conceptual guidelines or something similar